views/js/runner/testStore.js

```diff
@@ -0,0 +1,295 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA
+ *
+ */
+
+/**
+ * The test runner persistent data store,
+ * to be used with sub stores.
+ *
+ * Supports the legacy mode where multiple stores were used by each plugin.
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+define([
+    'lodash',
+    'core/promise',
+    'core/store',
+    'core/logger'
+], function (_, Promise, store, loggerFactory) {
+    'use strict';
+
+    /**
+     * The test store logger
+     * @type {core/logger}
+     */
+    var logger = loggerFactory('taoQtiTest/runner/provider/testStore');
+
+    /**
+     * Database name prefix (suffixed by the test identifier)
+     * to check if we use the legacy mode (multiple dbs)
+     * or the new mode (one db per test).
+     * @type {String[]}
+     */
+    var legacyPrefixes = [
+        'actions-', 'duration-', 'test-probe', 'timer-'
+    ];
+    /**
+     * Check and select the store mode.
+     * If any of the "legacyPrefixes" store is found, we used the "multiple/legacy" mode
+     * otherwise we use the unified mode.
+     * @param {String} testId
+     * @param {Object} [preselectedBackend] - the storage backend
+     * @returns {Promise<Boolean>} true means unified
+     */
+    var selectStoreMode = function selectStoreMode(testId, preselectedBackend){
+        return store
```

Comment:
I think this method could get a better name. We are not selecting anything here, we are just checking whether we have legacy-named stores for the current test. A "boolean-type" name seems more appropriated. Something like `hasLegacyStores` for example.